external-crates/move/crates/move-stackless-bytecode/src/conditions_order_analysis.rs

```diff
@@ -0,0 +1,376 @@
+                                                if param_type.is_mutable_reference() {
+                                                    results.insert(builder.get_loc(*attr)); 
+                                                }
+                                            }
+
+                                            println!("{:?} {} names", env.symbol_pool().string(module.get_name().name()), module.get_function(*fun_id).get_name_str());
+                                        },
+                                        _ => {}
+                                    };
+                                }
+                            },
+                            _ => {},
+                        }
+                    }
+                },
+            };                
+        }
+
+        results
+    }
+}
+
+impl FunctionTargetProcessor for ConditionOrderAnalysisProcessor {
+    fn process(
+        &self,
+        targets: &mut FunctionTargetsHolder,
+        func_env: &FunctionEnv,
+        data: FunctionData,
+        _scc_opt: Option<&[FunctionEnv]>,
+    ) -> FunctionData {
+        if !targets.is_spec(&func_env.get_qualified_id()) {
+            // only need to do this for spec functions
+            return data;
+        }
+
+        let env = func_env.module_env.env;
+        let func_target = FunctionTarget::new(func_env, &data);
+
+        let underlying_func_id = targets
+            .function_specs()
+            .iter()
+            .find_map(|(id, func)| (func_env.get_qualified_id() == *id).then_some(*func));
+
+        if underlying_func_id.is_none() {
+            env.diag(
+                Severity::Error,
+                &func_env.get_loc(),
+                "Spec underlying func is not found",
+            );
+
+            return  data;
+        }
+
+        let underlying_func = func_env.module_env.get_function(underlying_func_id.unwrap().id);
+
+        let spec_params = func_env.get_parameters();
+        let underlying_params = underlying_func.get_parameters();
+
+        if spec_params.len() != underlying_params.len() {
+            env.diag(
+                Severity::Error,
+                &func_env.get_loc(),
+                "Spec function have differ params count than underlying func",
+            );
+
+            return data;
+        }
+
+        for i in 0..spec_params.len() {
+            if spec_params[i].0 != underlying_params[i].0 {
+                env.diag(
+                    Severity::Warning,
+                    &func_env.get_loc(),
+                    "Spec function have differ params names than underlying func",
+                );
+            }
+
+            if spec_params[i].1 != underlying_params[i].1 {
+                env.diag(
+                    Severity::Error,
+                    &func_env.get_loc(),
+                    "Spec function have differ params type than underlying func",
+                );
+
+                return data;
+            }
+        }
```

Comment:
this checks that the 2 functions have the same signature. We should check that they have the same number or type parameters (this should be an error), and maybe that they have the same type parameter names (this should be a warning).